### ISSUE TITLE
perf: GridSettings.cells called very aggressively during interacting with the table

### ISSUE THREAD

**Issue by NoNameProvided** [handsontable/handsontable#5074]
### Description

The function returning the config of a cell is called very aggresively, when table is rendered or a user is interacting with the table.

```js
cells: (row?: number, col?: number, prop?: object)
``` 

This makes it impossible to do any calculation in that function as for a larger payloads that function is called more than 200000 times. 

### Steps to reproduce

1. Visit this provided Fiddle
2. Open the developer tools and see that that in a 4x10 table the `cells` function is called ~ 300 times
3. Clear console
4. Click in a cell
5. Observe that `cells` get called ~30 times for a single cell

### Observations

- the amount of calls depends on the width of the table
- it seems the clicked row is traversed two times when clicking a cell
- when the table is loaded it will be called for every data point even when virtualscroll is enabled, so it should not be called more than ~ 100 cell.

### Demo

https://jsfiddle.net/8ffpsqt6/17/

### Expected behaviour

- function is called once per cell when rendering or interacting with a cell
- function is not called for datapoints which are not rendered when virtualscroll is enabled

### Your environment
* Handsontable version: 2.0
* Browser Name and version: All
* Operating System: Windows and Mac

**Comment by AMBudnik** [comment]
Hi @NoNameProvided 

Thank you for doing this investigation.

I do not know if there is anything we can do to easily skip this calls but I agree that this can seriously slow down the performance with a call avalanche. 

We should discuss this subject while adding this changes https://github.com/handsontable/handsontable/issues/3276 as the subject are closely related.

**Comment by AMBudnik** [comment]
Related https://github.com/handsontable/handsontable/issues/5204

**Comment by AMBudnik** [comment]
This topic is the same as https://github.com/handsontable/handsontable/issues/5204 so I will update it once we get any new on https://github.com/handsontable/handsontable/issues/5204

**Comment by AMBudnik** [comment]
Hi @NoNameProvided 
The origin issue was just closed as solved https://github.com/handsontable/handsontable/issues/5204
This issue is no longer replicable in Handsontable 12.3.0

**Comment by NoNameProvided** [comment]
Hi @AMBudnik! 

I can still see high render counts. We are down from [300 renders to 40 in the example](https://jsfiddle.net/vdkhxbm1/1/) which is good progress. However, 40 is still more than the one render I would expect. 

Clicking any cell in the table and then clicking another will log 40 times. This means when a cell loses focus all other cell is re-rendered even when no content changes in them. 

The other topic mentioned in my issue still seems to be true: Handsontable doesn't respect virtual scroll when rendering and will call `cells` for cells not visible currently. 

In the [following example](https://jsfiddle.net/2r65tgem/), only 40 cells are visible at the time but Handsontable will call the `cells` function around 200 times when a single cell is clicked.

---

To sum up: this is way better than it used to be, but I still doubt this is usable with anything moderately complex in that function. The project I have worked on is not in active development 5 years later, so I have no real-life example to test these changes, but I am fairly sure that most developers would agree with me that there is definitely more space for improvement in this area.

**Comment by AMBudnik** [comment]
Thank you for the input @NoNameProvided 

Based on Handsontable is built, we cannot renderer fewer cells that are in the viewport. 
I will, however, discuss the case of https://jsfiddle.net/2r65tgem/ with the team and get back to leave a comment here in this issue.

And when it comes to this

> Clicking any cell in the table and then clicking another will log 40 times.

This is related to eco-rendering task reported at https://github.com/handsontable/handsontable/issues/5769, where we would like to rerender only altered cells, so it would 1 call if you edited a single cell.

**Comment by AMBudnik** [comment]
Hi @NoNameProvided 

At the moment there is no way to avoid `cells` being called. the `viewportRowRenderingOffset` option can help if you edit a cell, then for example we get `145` calls instead of `1000`. Same situation for scrolling row by row, it's around `140` calls. If you'd like to avoid calls for `cells` during scrolling you can use the `viewportRowRenderingOffset: countRows()` which is pretty much the same as `renderAllRows()` in this case. then you get `1000` calls for editing but `0` for scrolling.

ps. I will make sure to notify you once we start with the eco-renderers projects. That should make the biggest difference.
